Source/Particles/ParticleBoundaryBuffer.cpp

```diff
@@ -0,0 +1,200 @@
+                               << " for side " << iside << " of dim " << idim << "\n";
+            }
+        }
+    }
+#ifdef AMREX_USE_EB
+    auto& buffer = m_particle_containers[2*AMREX_SPACEDIM];
+    for (int i = 0; i < numSpecies(); ++i)
+    {
+        int np = buffer[i].isDefined() ? buffer[i].TotalNumberOfParticles(false) : 0;
+        amrex::Print() << "Species " << getSpeciesNames()[i] << " has "
+                       << np << " particles in the EB boundary buffer \n";
+    }
+#endif
+}
+
+void ParticleBoundaryBuffer::clearParticles () {
+    for (int i = 0; i < numBoundaries(); ++i)
+    {
+        auto& buffer = m_particle_containers[i];
+        if (buffer[i].isDefined())
+        {
+            buffer[i].clearParticles();
+        }
+    }
+}
+
+void ParticleBoundaryBuffer::gatherParticles (MultiParticleContainer& mypc,
+                                              const amrex::Vector<const amrex::MultiFab*>& distance_to_eb)
+{
+    using PIter = amrex::ParConstIter<0,0,PIdx::nattribs>;
+    const auto& warpx_instance = WarpX::GetInstance();
+    const amrex::Geometry& geom = warpx_instance.Geom(0);
+    auto plo = geom.ProbLoArray();
+    auto phi = geom.ProbHiArray();
+    auto dxi = geom.InvCellSizeArray();
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim)
+    {
+        if (geom.isPeriodic(idim)) continue;
+        for (int iside = 0; iside < 2; ++iside)
+        {
+            auto& buffer = m_particle_containers[2*idim+iside];
+            for (int i = 0; i < numSpecies(); ++i)
+            {
+                if (!m_do_boundary_buffer[2*idim+iside][i]) continue;
+                const auto& pc = mypc.GetParticleContainer(i);
+                if (!buffer[i].isDefined())
+                {
+                    buffer[i] = ParticleBuffer::getTmpPC<amrex::PinnedArenaAllocator>(&pc);
+                    buffer[i].AddIntComp(false);  // for timestamp
+                }
+                auto& species_buffer = buffer[i];
+                for (int lev = 0; lev < pc.numLevels(); ++lev)
+                {
+                    const auto& plevel = pc.GetParticles(lev);
+                    for(PIter pti(pc, lev); pti.isValid(); ++pti)
+                    {
+                        auto index = std::make_pair(pti.index(), pti.LocalTileIndex());
+                        if(plevel.find(index) == plevel.end()) continue;
+
+                        auto& ptile_buffer = species_buffer.DefineAndReturnParticleTile(
+                                                        lev, pti.index(), pti.LocalTileIndex());
+                        const auto& ptile = plevel.at(index);
+                        auto np = ptile.numParticles();
+                        if (np == 0) continue;
+
+                        auto dst_index = ptile_buffer.numParticles();
+                        ptile_buffer.resize(dst_index + np);
+
+                        int timestamp_index = ptile_buffer.NumRuntimeRealComps()-1;
```

Comment:
Should this be `int timestamp_index = ptile_buffer.NumIntComps()-1` after the switch to using the timestep? (and same on line 176)

Reply:
Applied in the latest version of the code.